File: MesEquipment.Web/src/app/models/machine.ts
export interface Machine {
  id: number;
  code: string;
  name: string;
  location: string;
  status: string;
  createdAt: string;
}

export interface CreateMachine {
  code: string;
  name: string;
  location: string;
  status: string;
}

export interface UpdateMachine {
  code: string;
  name: string;
  location: string;
  status: string;
}

export interface MachineQuery {
  page: number;
  pageSize: number;
  keyword?: string;
  status?: string;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
}

File: backend/MesEquipment.Api/Models/Machine.cs
namespace MesEquipment.Api.Models;

public class Machine
{
    public int Id { get; set; }

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Location { get; set; } = string.Empty;

    public string Status { get; set; } = "Idle";

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public List<MachineStatusLog> StatusLogs { get; set; } = [];
}